pkg/runtime/field_export_reconciler.go

```diff
@@ -0,0 +1,673 @@
+
+	if r.rd != nil {
+		return errors.New("cannot bind field export reconciler with non-nil resource descriptor")
+	}
+
+	return ctrlrt.NewControllerManagedBy(
+		mgr,
+	).For(
+		// Read only field export objects
+		&ackv1alpha1.FieldExport{},
+	).WithEventFilter(
+		predicate.GenerationChangedPredicate{},
+	).Complete(r)
+}
+
+// BindServiceResourceToManager binds a given AWS resource descriptor to the
+// controller manager
+func (r *fieldExportReconciler) BindServiceResourceManager(mgr ctrlrt.Manager) error {
+	r.kc = mgr.GetClient()
+	r.apiReader = mgr.GetAPIReader()
+
+	if r.rd == nil {
+		return errors.New("cannot bind resource field export reconciler with nil resource descriptor")
+	}
+
+	return ctrlrt.NewControllerManagedBy(
+		mgr,
+	).For(
+		(*r.rd).EmptyRuntimeObject(),
+	).WithEventFilter(
+		// Update on both status and spec changes
+		predicate.ResourceVersionChangedPredicate{},
+	).Complete(r)
+}
+
+// Reconcile implements `controller-runtime.Reconciler` and handles reconciling
+// a CR CRUD request
+func (r *fieldExportReconciler) Reconcile(ctx context.Context, req ctrlrt.Request) (ctrlrt.Result, error) {
+	return r.handleReconcileError(r.reconcile(ctx, req))
+}
+
+func (r *fieldExportReconciler) reconcile(ctx context.Context, req ctrlrt.Request) error {
+	// Determine if we are reconciling an ACK resource
+	if r.rd != nil {
+		return r.reconcileResource(ctx, req)
+	}
+
+	// We are reconciling a field export CR
+	return r.reconcileFieldExport(ctx, req)
+}
+
+// reconcileFieldExport handles updates to `FieldExport` resources
+func (r *fieldExportReconciler) reconcileFieldExport(ctx context.Context, req ctrlrt.Request) error {
+	res, err := r.getFieldExport(ctx, req)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// resource wasn't found. just ignore these.
+			return nil
+		}
+		return err
+	}
+
+	sourceGK := res.Spec.From.Resource.GroupKind
+	sourceName := types.NamespacedName{
+		Name: *res.Spec.From.Resource.Name,
+		// We only support pulling from resources in
+		// the same namespace
+		Namespace: req.Namespace,
+	}
+
+	// Check if the target API group matches with the controller
+	var controllerRMF acktypes.AWSResourceManagerFactory
+	for _, v := range r.sc.GetResourceManagerFactories() {
+		controllerRMF = v
+		break
+	}
+	if sourceGK.Group != controllerRMF.ResourceDescriptor().GroupKind().Group {
+		ackrtlog.DebugFieldExport(r.log, res, "target resource API group is not of this service. no-op")
+		return nil
+	}
+
+	if res.DeletionTimestamp != nil {
+		return r.cleanup(ctx, res)
+	}
+
+	if err := r.markManaged(ctx, res); err != nil {
+		return r.onError(ctx, res, err)
+	}
+
+	// Look up the rmf for the given target resource GVK
+	rmf, ok := (r.sc.GetResourceManagerFactories())[sourceGK.String()]
+	if !ok {
+		return ackerr.ResourceManagerFactoryNotFound
+	}
+
+	sourceObject, err := r.getSourceResource(ctx, rmf.ResourceDescriptor(), sourceName)
+	if err != nil {
+		return r.onError(ctx, res, err)
+	}
```

Comment:
> Is this the user experience we want?

Nope. In fact this is what I was trying to avoid, but I got caught up in the `err != nil` copypasta. I will replace this with a return success. No need to requeue it, since the field export reconciler will run again once the resource has been created and export it (now that we have marked this field export as managed).